### PR TITLE
Add `allowNonzeroExitCode` option

### LINKED ISSUE #86: reject on return code > 0
I have a problem with these lines:
```
if (code > 0) {
  reject(new Error('Exited with code ' + code));
  return;
}
```

This assumes that an exit code of > 0 signals an error and other exit codes (0 or negative) signal success.
For one thing: Yes, the exit code is a _signed_ integer and convention says everything other than 0 signals an error.
Second: It's purely a convention that an exit code of 0 signals success, not a standard. Depending on the OS and environment it could just as well signal true, false or "nothing has been processed". An exit code of 5 could signal error or it could signal "I have processed 5 items" or "main() had no return because the programming tutorial said the function signature is 'void main()' and 5 is what happened to be on the stack".

Especially on Windows far too many applications don't follow the 0=success convention and since opn opens a default application we can't know if the application that does get run follows convention, so what is the rejection good for
Essentially as soon as windows is one of your target platforms you have to catch and ignore rejection because there is a good chance it rejects for no reason, but thereby also ignore actual errors reported by the error callback.

### PR DESCRIPTION
**Pull request by peterchu999**
fixing issue using #161 as reference 

Fixes #86

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -42,6 +42,14 @@ declare namespace open {
 		@default false
 		*/
 		readonly url?: boolean;
+
+		/**
+		Use to allow non zero exit code. 
+		the convention for success exit code was 0, But in some cases there is an app that doesn't return 0 for the success exit code.
```
Comment (sindresorhus) [inline index.d.ts:48]: Put some more effort into this text.

File: index.d.ts [hunk 2]
```diff
@@ -42,6 +42,14 @@ declare namespace open {
...
+
+		@default true
```
Comment (sindresorhus) [inline index.d.ts:50]: It should not be `true` by default.

File: index.d.ts [hunk 3]
```diff
@@ -42,6 +42,14 @@ declare namespace open {
...
+		 */
```
Comment (sindresorhus) [inline index.d.ts:51]:
```suggestion
		*/
```

File: index.d.ts [hunk 4]
```diff
@@ -42,6 +42,14 @@ declare namespace open {
 		@default false
 		*/
 		readonly url?: boolean;
+
+		/**
+		Use to allow non zero exit code.
```
Comment (sindresorhus) [inline index.d.ts:47]:
```suggestion
		Use to allow nonzero exit code. 
```

**Comment by peterchu999** [comment]
okay i'll try to make it better

File: index.d.ts [hunk 5]
```diff
@@ -42,6 +42,15 @@ declare namespace open {
 		@default false
 		*/
 		readonly url?: boolean;
+
+		/**
+		Option to allow the opened app resolve the promise with nonzero exit code when `wait` option value `true`.
```
Comment (sindresorhus) [inline index.d.ts:47]: It's already clear from the context that this is an option.

File: index.d.ts [hunk 6]
```diff
@@ -42,6 +42,15 @@ declare namespace open {
...
+		
+		We do not recommend using it because by default the app convention for success exit code was 0.
```
Comment (sindresorhus) [inline index.d.ts:49]: was => is

File: index.d.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /// <reference types="node"/>
-import {ChildProcess} from 'child_process';
+import { ChildProcess } from "child_process";
```
Comment (sindresorhus) [inline index.d.ts:2]: Don't do unrelated changes.

Comment (peterchu999) [reply index.d.ts:2]: sorry this was due to my formatter

File: readme.md [hunk 8]
```diff
@@ -102,6 +102,14 @@ We do not recommend using it on targets that are not URLs.
 
 Especially useful when dealing with the [double-quotes on Windows](#double-quotes-on-windows) caveat.
 
+##### allowNonzeroExitCode
+
+Type: `boolean`\
+Default: `false`
+
+Option to allow the opened app resolve the promise with nonzero exit code when `wait` option value `true`.
```
Comment (sindresorhus) [inline readme.md:110]: The readme docs should be in sync with the index.d.ts. The text here is different.

Comment (peterchu999) [reply readme.md:110]: so sorry i miss the README
